### PR TITLE
Updating tests

### PR DESCRIPTION
**Pull request by hamshavathimunibyraiah**
Updating tests:

1. Authenticating storage accounts using MI
2. SAS urls are no longer needed to access containers using DefaultAzureCredential instead
3. Updated test-resource.json inline with what we have for other langs
4. Updating resource output variable names accordingly.
5. Updated text translation test cases: updated resource output names
6. Added test cases for the overridden models and operations (startTranslation using start_translation_details)

### CODE REVIEW

File: sdk/translation/tests.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ extends:
     template: /eng/pipelines/templates/stages/archetype-sdk-tests.yml
     parameters:
       ServiceDirectory: translation
+      UseFederatedAuth: true
```
[the remark this answers is not in the document]
Comment (weshaggard) [reply sdk/translation/tests.yml:7]: Correct this shouldn't be needed.

[the remark this answers is not in the document]
Comment (hamshavathimunibyraiah) [reply sdk/translation/tests.yml]: Removed

**Pull request closed without merging**